snuba/datasets/entities/metrics.py

```diff
@@ -77,16 +77,27 @@ def transform_expression(exp: Expression) -> Expression:
 class MetricsEntity(Entity, ABC):
     def __init__(
         self,
-        writable_storage_key: StorageKey,
+        writable_storage_key: Optional[StorageKey],
         readable_storage_key: StorageKey,
         value_schema: Sequence[Column[SchemaModifiers]],
         mappers: TranslationMappers,
+        allow_cross_org: bool = False,
     ) -> None:
-        writable_storage = get_writable_storage(writable_storage_key)
+        writable_storage = (
+            get_writable_storage(writable_storage_key) if writable_storage_key else None
+        )
         readable_storage = get_storage(readable_storage_key)
+        storages = [readable_storage]
+        if writable_storage:
+            storages.append(writable_storage)
+
+        validators = []
+        if not allow_cross_org:
+            validators.append(EntityRequiredColumnValidator({"org_id", "project_id"}))
+        validators.append(GranularityValidator(minimum=10))
```

Comment:
Not sure if order matters here, otherwise I can just include `GranularityValidator` in the initial list